### ISSUE TITLE
kiosk in kconfig

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/kconfig#10]
- [ ] document metadata in doc/METADATA.ini (@darddan), ideally using `restrict/write`
- [ ] implement in qt-gui
- [ ] implement kconfig (handle metadata) (@FelixResch)

**Comment by darddan** [comment]
about KConfig kiosk mode:
if a group ends with `[$i]` (which is the only meta allowed at groups, it is considered as an immutable group. What I didn't notice earlier, if this is added at an empty line, without a group name, it makes all the following groups immutable.

When a group is immutable, you can't add, remove or edit children. In Elektra, we should restrict to add/remove/edit immediate  "children". 

As far as I could understand. If we the `restrict/write` meta of a key to 1, we can't change this key itself, which isn't the same functionality. Should I add another meta `immediate/children/restrict/write` or something?

Also there is a global variable that disables the kiosk mode functionality, which could be added as `debug mode` in the qt-gui (similar to the `Viewermode`)

**Comment by darddan** [comment]
To explain the `immediate children` logic, If we had a config file:

```
[group]
subgroup=1
[group][subgroup][$i]
entry=2
[group][subgroup][subsubgroup]
entry=3
```
* The `kdb set group/subgroup 5` should work. Even though both the entry and the restricted group have the same path (`group/subgroup`), the entry is considered to be in another group.
* The `kdb set group/subgroup/entry 5` shouldn't work because the group is immutable.
* The `kdb set group/subgroup/subsubgroup/entry 6` should work, even though in elektra is mapped under an immutable "node".

**Comment by markus2330** [comment]
> As far as I could understand. If we the restrict/write meta of a key to 1, we can't change this key itself, which isn't the same functionality. Should I add another meta immediate/children/restrict/write or something?

Yes, such an extension will be required. But the prefix should stay the same so that similar meta data stays together. What about the name `restrict/write/children/immediate`?

Note that actually also for the currently defined `restrict` metadata no plugins exist. So your kconfig plugin needs not only to add the correct metadata when it finds `$i` but we also need a plugin that actually enforces all these restrictions.
